### DISCUSSION TITLE
Week 12 Report

### DISCUSSION THREAD

**Discussion by derrrobin** [SE-TINF22B6/CookPit#72] [General]
# Weekly Blog: CookPit Project Update

This week at CookPit, our team dedicated significant effort to enhancing the reliability and user experience of our platform. We have worked on the website layout for mobile users and tried to fix many bugs. We also added the imprint and the FAQs. 

<img width="490" alt="image" src="https://github.com/SE-TINF22B6/CookPit/assets/123726577/ae35c102-8396-487c-8504-aa8a56bc6be5">

One of the pivotal steps we took was to revamp our testing strategy and documentation, ensuring thorough validation of our software.

## Test Strategy
Our testing strategy is comprehensive, integrating various methodologies to ensure the robustness of our software. We've adopted a cyclical testing approach, embedding testing activities throughout the development cycle to catch and address issues early on.

## Test Plan
Our test plan encompasses different types of tests to validate critical aspects of our platform:

- **Unit Testing:** Rigorous testing of React components using JUnit.
- **Integration Testing:** Ensuring seamless interaction between components with tools like Jest and Enzyme.
- **Validation Testing:** Verifying adherence to requirements and user expectations through manual and automated validation.
- **Backend Testing:** Employing GitHub Actions for automated backend testing to ensure stability.
- **OpenAI API Testing:** Thorough validation of our integration with the OpenAI API.

## Automatic Testing Tools
We've integrated GitHub Actions into our workflow for automated testing after each release, ensuring continuous quality assurance.

## Test Case Management
Our test cases are meticulously organized by functionality, documented with clear objectives, steps, and expected results. Version control and traceability links ensure comprehensive coverage, while systematic recording of test results facilitates insightful reporting.

-> To sum up
The enhancements made to our testing process this week signify our unwavering commitment to delivering a reliable and user-friendly platform. By prioritizing thorough testing and documentation, we're poised to provide our users with a seamless cooking experience. Stay tuned for further updates on our progress!

**Comment by Stev3eM** [comment]
Hey CookPit Team,

Nice work on the latest updates! It's impressive seeing you focus not just on fixing bugs but also on making sure everything runs smoothly through testing (Suprise suprise, we will make some tests too 🥼). Your approach with GitHub Actions for automation and detailed test plans really shows a commitment to quality.

Looking forward to see more from you guys and how all these improvements play out for users. Maybe we can make a small cook battle in the end when the projects finally over. 

See you next time!

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**

**Comment by nuo-li** [comment]
Dear CookPit team, would you please introduce a bit about Enzyme? Why do you choose it?

**Reply by derrrobin** [reply]
Hi, using Enzyme was just one idea of us, because its made for testing react components which is our use case. 

Here are some highlights for our team, why to us Enzyme:

1. Enzyme provides rendering capabilities, allowing you to isolate and test individual components without rendering child components
2.   We can alsou simulate user interactions such as clicks, inputs, and form submissions. This facilitates comprehensive testing of user interactions and ensures that components respond correctly to user actions. (Login, Upload recipe, Favorites)
3. Also a point why we named Enzyme is, that we are using React for the first Time, and Enzyme is widely adopted within the React community and has extensive documentation and support resources available.
